gen/enum.go

```diff
@@ -101,6 +106,99 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 			<end>
 			return fmt.Sprintf("<$enumName>(%d)", <$w>)
 		}
+
+		func (<$v> <$enumName>) MarshalJSON() ([]byte, error) {
+			<if len .Spec.Items>
+				<$w> := int32(<$v>)
+				switch <$w> {
+				<range .UniqueItems>
+					case <.Value>:
+						return ([]byte)("\"<.Name>\""), nil
+				<end>
+				}
+			<end>
+			return ([]byte)(<$strconv>.FormatInt(int64(<$v>), 10)), nil
+		}
+
+		func (<$v> *<$enumName>) UnmarshalJSON(text []byte) error {
+			<$d := newVar "d">
+			<$t := newVar "t">
+			<$err := newVar "err">
```

Comment:
Don't use `newVar` for err. That isn't done anywhere else. Assumption is that err is always shadowed.